### PR TITLE
[ui] Enable displaying recently visited assets

### PR DESCRIPTION
**Pull request by clairelin135**
Creates a LRU cache of the last 10 visited asset keys, storing them in local storage.

Displays the recently visited assets on the asset catalog page in chronological order of visit.

### CODE REVIEW

**Comment by clairelin135** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20266?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#20266** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20266?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#20210** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20210?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clairelin135 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: js_modules/dagster-ui/packages/ui-core/src/assets/RecentlyVisitedAssetsStorage.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import {AssetKey} from './types';
+
+const RECENTLY_VISITED_ASSETS_CACHE_SIZE = 10;
+const RECENTLY_VISITED_ASSETS_STORAGE_KEY = 'recentlyVisitedAssets';
+
+export function WriteAssetVisitToLocalStorage(assetKey: AssetKey) {
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/RecentlyVisitedAssetsStorage.tsx:6]: `camelCase` function names unless its a function component.

File: js_modules/dagster-ui/packages/ui-core/src/assets/RecentlyVisitedAssetsStorage.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    const truncatedVisitedAssets = visitedAssets.slice(0, RECENTLY_VISITED_ASSETS_CACHE_SIZE);
+    localStorage.setItem(
+      RECENTLY_VISITED_ASSETS_STORAGE_KEY,
+      JSON.stringify(truncatedVisitedAssets),
+    );
+  }
+}
+
+export function FetchRecentlyVisitedAssetsFromLocalStorage(): AssetKey[] {
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/RecentlyVisitedAssetsStorage.tsx:32]: same

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogRoot.tsx [hunk 3]
```diff
@@ -82,6 +83,10 @@ export const AssetsCatalogRoot = () => {
     );
   }
 
+  // If the asset exists, add it to the recently visited list
+  if (currentPath && currentPath.length) {
+    WriteAssetVisitToLocalStorage({path: currentPath});
+  }
```
[the remark this answers is not in the document]
Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogRoot.tsx]: thanks, adjusted this

**Review by salazarm — APPROVED** [review]
This looks good to me!

**Pull request merged**